File: SmartValley.WebApi/Projects/Responses/ScoringProjectResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using SmartValley.WebApi.Scorings;

namespace SmartValley.WebApi.Projects.Responses
{
    public class ScoringProjectResponse
    {
        public long ProjectId { get; set; }
        
        public string ProjectExternalId { get; set; }

        public string Address { get; set; }

        public string Name { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public ScoringProjectStatus Status { get; set; }

        public IEnumerable<AreaExpertResponse> AreasExperts { get; set; }

        public static ScoringProjectResponse Create(ScoringProjectDetailsWithCounts details)
        {
            return new ScoringProjectResponse
                   {
                       Address = details.ContractAddress,
                       Name = details.Name,
                       ProjectId = details.ProjectId,
                       ProjectExternalId = details.ProjectExternalId.ToString(),
                       StartDate = details.CreationDate.Date,
                       EndDate = details.AcceptingDeadline.Date,
                       Status = details.Status,
                       AreasExperts = details.AreaCounts.Select(AreaExpertResponse.Create).ToArray()
                   };
        }
    }
}

File: SmartValley.FrontEnd/src/app/api/scoring-application/scoring-project-response.ts
import {ScoringProjectStatus} from '../../services/scoring-project-status.enum';
import {AreaExpertResponse} from '../project/area-expert-response';

export interface ScoringProjectResponse {
  projectId: number;
  projectExternalId: string;
  name: string;
  address: string;
  startDate?: Date;
  endDate?: Date;
  status: ScoringProjectStatus;
  areasExperts: Array<AreaExpertResponse>;
}
